tests/unit/test_rundb.py

```diff
@@ -175,3 +176,81 @@ def callback(index, row):
         match='not defined.*\n.*foo.*\n.*bar'
     ):
         dao.select_task_pool_for_restart(callback)
+
+
+@pytest.fixture(scope='module')
+def setup_message_in_db(tmp_path_factory):
+    """Create a database for use when testing message_in_db"""
+    tmp_path = tmp_path_factory.mktemp('message_in_db')
+    db_file = tmp_path / 'db'
+    dao = CylcWorkflowDAO(db_file, create_tables=True)
+    setup_stmt = r"""
+        INSERT INTO task_events
+        VALUES
+            (
+                "qux", "-191",
+                "morning", "1", "started", ""),
+            (
+                "qux", "-190",
+                "afternoon", "1", "message critical", "Hello Rome");
+    """
+    dao.connect().execute(setup_stmt)
+    yield dao
```

Comment:
Closing DB may avoid any possible flakiness
```suggestion
    setup_stmt = r"""
        INSERT INTO task_events
        VALUES
            (
                "qux", "-191",
                "morning", "1", "started", ""),
            (
                "qux", "-190",
                "afternoon", "1", "message critical", "Hello Rome");
    """
    with CylcWorkflowDAO(db_file, create_tables=True) as dao:
        dao.connect().execute(setup_stmt)
        yield dao
```